### PR TITLE
DS-690: Remove Nightwatch

### PR DESCRIPTION
**Pull request by colbytcook**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-690

## Summary

Removed Nightwatch from the build.

## Details

* Removed Nightwatch tasks from the main `package.json` file
* Removed the Nightwatch tasks from the `travis.xml` file
* Deleted all the `.e2e.js` files within the `/packages/components` folder
* Removed Nightwatch dependencies
* Removed the `packages/testing/ttesting-nightwatch` folder

## How to test

1. Review the code
2. Confirm that Travis has successfully ran (with no Nightwatch tasks running)
3. Confirm that Travis has successfully ran on this https://github.com/boltdesignsystem/bolt/pull/2431 (with no Nightwatch tasks running)

### CODE REVIEW

**Review by danielamorse — APPROVED** [review]
@colbytcook this looks great. The main things I wanted to confirm were 1) There were no more Nightwatch remnants (none found) 2) There were no unnecessary Travis build steps once Nightwatch was removed (looks good!).

Good to merge once conflicts are resolved and passing on Travis 🚀

File: jest.config.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // https://facebook.github.io/jest/docs/en/configuration.html
 const globby = require('globby');
 const testFilesToIgnore = globby.sync([
-  'packages/components/**/*.e2e.js',
-  'packages/components/**/*.no2e.js', // workaround to disable a few e2e tests
+  // 'packages/components/**/*.e2e.js',
```
Comment (adamszalapski) [inline jest.config.js:4]: I suggest removing this line 4 and 5, and we should check if all e2e and no2e files are removed from the repo.

**Review by adamszalapski — CHANGES_REQUESTED** [review]
I noticed that in `.gitignore` we have some info about nightwatch
```
# nightwatch.js test results when run locally
/tests_output/**
tests_output
```

**Review by remydenton — APPROVED** [review]
The code here looks good to me.

I think I missed the convo where we decided to remove Nightwatch though.  Regarding thd description in JIRA the ticket:

> Since we have a QA team that is performing end to end testing...

Do we actually have a QA team doing that in Bolt?  And even if so, do they have the time to be as thorough as an automated test suite?

I trust you guys have already thought about this and made the right call, just noting here that these kinds of tests can indeed provide value.

File: .travis.yml [hunk 2]
```diff
@@ -138,7 +137,7 @@ jobs:
         - export NOW_URL=$(./scripts/get-latest-deploy.js)
         - travis_retry yarn gds -a create -e ${TRAVIS_BRANCH}--branch-preview -r ${TRAVIS_BRANCH}
         - travis_retry yarn gds -a in_progress -e ${TRAVIS_BRANCH}--branch-preview -r ${TRAVIS_BRANCH}
-        - travis_retry yarn run test:e2e:full
+        # - travis_retry yarn run test:e2e:full
```
[the remark this answers is not in the document]
Comment (colbytcook) [reply .travis.yml]: @adamszalapski this has been updated

[the remark this answers is not in the document]
Comment (colbytcook) [reply .travis.yml]: @adamszalapski this has been updated

**Review by adamszalapski — APPROVED** [review]

**Pull request merged**